src/parser/Parser.ts

```diff
@@ -746,11 +755,40 @@ export class Parser {
     }
 
     private fieldDeclaration(accessModifier: Token | null) {
-        let name = this.consume(
-            DiagnosticMessages.expectedClassFieldIdentifier(),
-            TokenKind.Identifier,
-            ...AllowedProperties
-        ) as Identifier;
+
+        let optionalKeyword = this.consumeTokenIf(TokenKind.Optional);
+
+        let name: Identifier;
+
+        if (this.checkAny(TokenKind.Identifier, ...AllowedProperties)) {
+            if (this.check(TokenKind.As)) {
+                if (this.checkAnyNext(TokenKind.Comment, TokenKind.Newline)) {
+                    // as <EOL>
+                    // `as` is the field name
+                } else if (this.checkNext(TokenKind.As)) {
+                    //  as as ____
+                    // first `as` is the field name
+                } else if (optionalKeyword) {
+                    // optional as ____
+                    // optional is the field name, `as` starts type
+                    // rewind current token
+                    optionalKeyword = null;
+                    this.current--;
+                }
+            }
+
+            name = this.consume(
+                DiagnosticMessages.expectedClassFieldIdentifier(),
+                TokenKind.Identifier,
+                ...AllowedProperties
+            ) as Identifier;
+        } else {
+            // no name after `optional` ... optional is the name
+            name = optionalKeyword as Identifier;
```

Comment:
Since `optional` is the name of the field, I think we need to change the optionalKeyword.kind to `Identifier` instead of just casting it in typescript land. 

Like this:
https://github.com/rokucommunity/brighterscript/blob/5a68874bd86dbf88c88db8a0a545b4b00b5db8dd/src/parser/Parser.ts#L361